rendermodules/dataimport/apply_mipmaps_to_render.py

```diff
@@ -102,7 +102,22 @@ def run(self):
 
         self.output_tilespecs_to_stack(tilespecs, output_stack,
                                        sharedTransforms=identified_tforms)
-        self.output({"output_stack": output_stack})
+
+        missing_ts_zs = self.validate_apply_mipmaps(self.args['input_stack'], output_stack, zvalues)
+
+        self.output({"output_stack": output_stack, "missing_tilespecs_zs": missing_ts_zs})
+
+    def validate_apply_mipmaps(self, input_stack, output_stack, zvalues):
+        missing_ts_zs = []
+        
+        for z in zvalues:
+            input_rs = renderapi.resolvedtiles.get_resolved_tiles_from_z(input_stack, z, render=self.render)
+            output_rs = renderapi.resolvedtiles.get_resolved_tiles_from_z(output_stack, z, render=self.render)
+            if len(input_rs.tilespecs) != len(output_rs.tilespecs): # pragma: no cover
```

Comment:
why not raise an exception here and have the job fail? what happens here is that there are missing tiles and it is just logged to the output json.

Reply:
the json output was for the workflow to catch it and set the job state appropriately. raising a validation error after the first missing tileset would not allow for us to check the entire list of Zs. This behavior is not desired.